backend/middleware/authMiddleware.js

```diff
@@ -0,0 +1,16 @@
+import { verify } from "jsonwebtoken";
+
+const authenticateToken = (req, res, next) => {
+  const token = req.header("Authorization")?.split(" ")[1];
+  if (!token) return res.status(401).json({ message: "Access denied" });
+
+  try {
+    const verified = verify(token, process.env.JWT_SECRET);
+    req.user = verified;
+    next();
+  } catch (error) {
+    res.status(400).json({ message: "Invalid token" });
```

Comment:
Would it be better to return 401 here (https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/401)?

Reply:
You're right. 401 would be a better code for this situation